### PR TITLE
236 milestone take the un squared version of the current z statistics

### LINKED ISSUE #236: `milestone()`: take the un-squared version of the current Z statistics
The equation (3) of Klein, et al (2007) published at Stat in Medicine has a squared version of the test statistics. We will take the un-squared version to provide a directional Z test.
image

### PR DESCRIPTION
**Pull request by LittleBeannie**
Closes #236

### CODE REVIEW

**Review by nanxstats — APPROVED** [review]
I see a unit test is skipped for now - would be great to unskip it later after the hypothesis testing syntax is updated.

**Comment by jdblischak** [comment]
> I see the "multitest" unit test is skipped for now - would be great to unskip it later after the hypothesis testing syntax is updated.

I don't understand why this test was skipped. It doesn't run the updated `milestone()` function, so it should have been unaffected by this PR. I just ran it locally, and it passed

**Comment by nanxstats** [comment]
> I don't understand why this test was skipped. It doesn't run the updated `milestone()` function, so it should have been unaffected by this PR. I just ran it locally, and it passed

🤔 That would be a good question for @LittleBeannie

File: R/milestone.R [hunk 1]
```diff
@@ -23,13 +23,18 @@
 #'   - `event` - Event indicator.
 #'   - `treatment` - Grouping variable.
 #' @param ms_time Milestone analysis time.
-#'
+#' @param test_type Method to build the test statistics.
+#' There are 2 options:
+#'   - `"native"`: a native approach by dividing the KM survival difference by its standard derivations,
```
Comment (jdblischak) [inline R/milestone.R:28]: Is the approach "native" or "naive"? The string passed to `test_type` below is `"naive"`

File: R/milestone.R [hunk 2]
```diff
@@ -23,13 +23,18 @@
...
 #' @return A list frame containing:
-#' - `method` - The method, always `"milestone"`.
-#' - `parameter` - Milestone time point.
-#' - `estimation` - Survival difference between the experimental and control arm.
-#' - `se` - Standard error of the control and experimental arm.
-#' - `z` - Test statistics.
+#'   - `method` - The method, always `"milestone"`.
+#'   - `parameter` - Milestone time point.
+#'   - `estimation` - Survival difference between the experimental and control arm.
+#'   - `se` - Standard error of the control and experimental arm.
```
Comment (jdblischak) [inline R/milestone.R:36]: The `se` result was changed by this PR, but the documentation did not change. Previously it returned two standard error values (from `fit_res$std.err`), but now it only returns a single standard error. Is is the standard error of the control arm, experimental arm, or some combination? We should update the documentation for this return value

**Review by jdblischak — COMMENTED** [review]
Overall this PR had a much bigger impact than I had anticipated. I assumed it was only changing the Z statistics in `z`. However, it also changed `se` to only return a single value instead of two. And the default `estimate` is also changed by the new `test_type` of `"log-log"`. I was able to fix `estimate` for my `sim_gs_n()` tests by adding `test_type = "naive"`.

We should add some developer tests for `milestone()` so that it is more transparent how a PR is effecting the output.

File: R/milestone.R (hunk 1 above)
Comment (nanxstats) [reply R/milestone.R:28]: Probably a typo - should be "naive" cc @LittleBeannie

File: R/milestone.R (hunk 2 above)
Comment (nanxstats) [reply R/milestone.R:36]: Agreed, this should be a quick fix for @LittleBeannie

**Pull request merged**